priv/peg_includes.hrl

```diff
@@ -181,10 +181,12 @@ p_scan(P, Inp, Index, Accum) ->
 -ifdef(p_string).
 -spec p_string(binary()) -> parse_fun().
 p_string(S) ->
-    Length = erlang:byte_size(S),
+    Chars = string:length(S),
     fun(Input, Index) ->
       try
-          <<S:Length/binary, Rest/binary>> = Input,
+          S = string:slice(Input, 0, Chars),
+          Length = byte_size(S),
+          <<_:Length/binary, Rest/binary>> = Input,
```

Comment:
It'd be good to have a comment for why the slicing is handled this way, but not necessary.